common/src/main/java/io/pravega/common/util/ConfigurationOptionsExtractor.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.util;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A utility class for processing configuration items specified via system properties and/ environment variables.
+ */
+@Slf4j
+public class ConfigurationOptionsExtractor {
+
+    /**
+     * Extracts a configuration value as String from the specified from system property or environment variable. If
+     * both are specified, the value from system property is returned. If neither are specified, it returns
+     * the {code defaultValue}.
+     *
+     * @param systemProperty the system property to extract the configuration value from
+     * @param environmentVariable the environment variable to extract the configuration value from
+     * @param defaultValue the value to be returned if neither {@code systemProperty} or {@code environmentVariable} is
+     *                     specified
+     * @return the specified or default configuration value
+     */
+    public static String extractString(@NonNull String systemProperty, @NonNull String environmentVariable,
```

Comment:
Is this used anywhere? Can it be made private?

Reply:
I've reduced the visibility as suggested.